shared/src/types/ethereum_events/vote_extensions.rs

```diff
@@ -0,0 +1,299 @@
+        );
+        assert!(
+            FractionalVotingPower::new(1, 3).unwrap()
+                == FractionalVotingPower::new(2, 6).unwrap()
+        );
+    }
+
+    /// Test error handling on the FractionalVotingPower type
+    #[test]
+    fn test_fractional_voting_power_valid_fractions() {
+        assert!(FractionalVotingPower::new(0, 0).is_err());
+        assert!(FractionalVotingPower::new(1, 0).is_err());
+        assert!(FractionalVotingPower::new(0, 1).is_ok());
+        assert!(FractionalVotingPower::new(1, 1).is_ok());
+        assert!(FractionalVotingPower::new(1, 2).is_ok());
+        assert!(FractionalVotingPower::new(3, 2).is_err());
+    }
+
+    /// Test decompression of a set of Ethereum events
+    #[test]
+    fn test_decompress_ethereum_events() {
+        // we need to construct a `Vec<Signed<VoteExtension>>`
+        let sk_1 = key::testing::keypair_1();
+        let sk_2 = key::testing::keypair_2();
+
+        let last_block_height = BlockHeight(123);
+
+        let ev_1 = EthereumEvent::TransfersToNamada {
+            nonce: 1u64.into(),
+            transfers: vec![],
+        };
+        let ev_2 = EthereumEvent::TransfersToEthereum {
+            nonce: 2u64.into(),
+            transfers: vec![],
+        };
+
+        let ext = {
+            let mut ext = VoteExtension::empty(last_block_height);
+
+            ext.ethereum_events.push(ev_1.clone());
+            ext.ethereum_events.push(ev_2.clone());
+            ext.ethereum_events.sort();
+
+            ext
+        };
+
+        // assume both v1 and v2 saw the same events,
+        // so each of them signs `ext` with their respective sk
+        let ext_1 = Signed::new(&sk_1, ext.clone());
+        let ext_2 = Signed::new(&sk_2, ext);
+
+        let ext = vec![ext_1, ext_2];
+
+        // we have the `Signed<VoteExtension>` instances we need,
+        // let us now compress them into a single `VoteExtensionDigest`
+        let signatures: Vec<(_, Address)> = vec![
+            (ext[0].sig.clone(), (&sk_1.ref_to()).into()),
+            (ext[1].sig.clone(), (&sk_2.ref_to()).into()),
+        ];
+        let signers = {
+            let mut s = HashSet::new();
+            s.insert(signatures[0].1.clone());
+            s.insert(signatures[1].1.clone());
+            s
+        };
+        let events = vec![
+            MultiSignedEthEvent {
+                event: ev_1.clone(),
+                signers: signers.clone(),
+            },
+            MultiSignedEthEvent {
+                event: ev_2.clone(),
+                signers,
+            },
+        ];
+
+        let digest = VoteExtensionDigest {
+            events,
+            signatures,
+            nulls: HashSet::new(),
+        };
+
+        // finally, decompress the `VoteExtensionDigest` back into a
+        // `Vec<Signed<VoteExtension>>`
+        let decompressed = digest.decompress(last_block_height);
+
+        assert_eq!(ext, decompressed);
```

Comment:
For complete sanity's sake, can we test that that signatures for each `Signed<VoteExtension>` from the decompressed verifies correctly?
```
assert!(decompressed[0].verify(&sk_1.ref_to()).is_ok());
assert!(decompressed[1].verify(&sk_2.ref_to()).is_ok());
```